### PR TITLE
Store: Add UI state for reviews list

### LINKED ISSUE #17041: Store: Product Reviews: Add state infrastructure for listing and filtering reviews by status
We need state code to make it possible to fetch reviews into state. 

It should also be able to accept a product ID or look against all products. It should be filterable by status, accept a query/search string, and a pagination page number.

This should look similar to the order or product listing code.

### PR DESCRIPTION
**Pull request by justinshreve**
This PR adds an action, some reducers, and selectors for the reviews list, which allows us to track the current page, product (if the request is filtering by product), and search term being viewed.

I decided not to break this PR down since most of the lines come from tests or the README, but I can if size is a problem.

Closes #17041.

To Test:
* Run `npm run test-client client/extensions/woocommerce` and make sure all tests pass.

### CODE REVIEW

**Comment by ryelle** [comment]
Are reviews going to be editable? One thing I've just run into with #17883 is that now that I want to track order editing, I need to put the UI reducer into a subtree of `ui.orders` so that I can also include and `edits` tree.

**Comment by justinshreve** [comment]
Good call. For the initial version we will at least need to hold edit state for status, but this could easily expand in the future. I'll do something similar tomorrow.

**Comment by justinshreve** [comment]
@ryelle I've updated my PR here to match what you've done over at #17883.

File: client/extensions/woocommerce/state/ui/reviews/list/reducer.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+export function currentProduct( state = null, action ) {
+	const { type, query } = action;
+	if ( WOOCOMMERCE_UI_REVIEWS_SET_QUERY === type ) {
+		return query && query.product ? query.product : state;
+	}
+	return state;
+}
+
+/**
+ * Tracks the current page of reviews displayed for the current site.
```
Comment (ryelle) [inline client/extensions/woocommerce/state/ui/reviews/list/reducer.js]: `current page` -> `current search term`

Comment (justinshreve) [reply client/extensions/woocommerce/state/ui/reviews/list/reducer.js]: 😆

**Pull request merged**